components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/util/CORSConfigurationUtil.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ * NOTE: The code/logic in this class is copied from https://bitbucket.org/thetransactioncompany/cors-filter.
+ * All credits goes to the original authors of the project https://bitbucket.org/thetransactioncompany/cors-filter.
+ */
+
+package org.wso2.carbon.identity.cors.mgt.core.internal.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.base.IdentityConstants;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.cors.mgt.core.constant.ErrorMessages;
+import org.wso2.carbon.identity.cors.mgt.core.model.CORSConfiguration;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Utility class for CORS configuration operations.
+ */
+public class CORSConfigurationUtil {
+
+    /**
+     * Private constructor of CORSConfigurationUtil.
+     */
+    private CORSConfigurationUtil() {
+
+    }
+
+    /**
+     * Read the default CORS configuration properties in the identity.xml.
+     *
+     * @return Server default {@code CORSConfiguration} object.
+     */
+    public static CORSConfiguration getServerCORSConfiguration() {
```

Comment:
Since all the attributes of this configuration are read from the identity.xml, shouldn't this be a one time task?

Reply:
Moved the first read to a static variable.